docs/grammar/BallerinaParser.g4

```diff
@@ -423,7 +422,15 @@ actionInvocationStatement
     ;
 
 transactionStatement
-    :   TRANSACTION LEFT_BRACE statement* RIGHT_BRACE transactionHandlers
+    :   TRANSACTION (WITH transactionPropertyInitExpressionList)? LEFT_BRACE statement* RIGHT_BRACE transactionHandlers
+    ;
+
+transactionPropertyInitExpression
```

Comment:
Additional grammar abstraction is used since in future more QoS attributes (ex: includeCache(YES) ) can be added to the initial property list. Renamed the "...Expression" in the grammar file to the "...Statement" . Thanks